### PR TITLE
OAI: add support to extract embargo from bitstreams 

### LINKED ISSUE #8923: OAI endpoint no longer supports exposing embargo dates
**Is your feature request related to a problem? Please describe.**
In DSpace 5.x and 6.x, embargo dates where present in the item metadata, and therefore could be crosswalked and exposed in the OAI metadata. This is something particularly important when exposing the open access status of fulltext bitstreams in DSpace.
**Describe the solution you'd like**
For this information to be exposed back in the OAI metadata.

**Describe alternatives or workarounds you've considered**
We've considered using the open access status (open access, embargoed, etc.) instead but this not provide any information about when the file will become available and it is needed for a number of external open access services.

**Additional context**

### PR DESCRIPTION
**Pull request by amgciadev**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #8923 (if this fixes an issue ticket)
## Description
Added support in ItemUtils to extract embargo for bitstreams and expose it as part of bundles/file metadata.
uketd_dc.xsl has been amended to use the bitstream level embargo information

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* ItemUtils.java (dspace-oai) modified to include an addEmbargoField method
* uketd_dc.xsl to add the embargo information to the relevant schema field

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [X] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [X] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [X] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [X] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by bram-atmire** [comment]
this might be a stupid question that can be clarified with some sample output: let's say you have two bitstreams on an item, each with a different embargo date, is it then still clear in the OAI output which date goes with which file?

Or do you end up with two embargo dates for the item, without a link back to the specific bitstream?

The fact that DSpace can support multiple bitstreams in one item always makes these kinds of use cases tricky.

**Comment by amgciadev** [comment]
> this might be a stupid question that can be clarified with some sample output: let's say you have two bitstreams on an item, each with a different embargo date, is it then still clear in the OAI output which date goes with which file?
> 
> Or do you end up with two embargo dates for the item, without a link back to the specific bitstream?
> 
> The fact that DSpace can support multiple bitstreams in one item always makes these kinds of use cases tricky.

@bram-atmire your comment is quite timely! In the PR I've just ported the original method we had in 5.x but I am reviewing it now to properly account for cases where there are multiple relevant policies (i.e. multiple anonymous read and an active start date): I suggest that only one embargo is added, and this is the longest embargo to be on the safe side. I am currently testing the method and will commit the changes soon

**Comment by amgciadev** [comment]
Example OAI output (uketd_dc metadata format) with two bitstreams, both with embargoes.

`<dc:identifier xsi:type="dcterms:URI">http://localhost:4000/bitstreams/5ab7e71a-d4b5-4085-8f38-96d3cf24f987/download</dc:identifier>
   <uketdterms:checksum xsi:type="uketdterms:MD5">d0b058ba9a2af50f0fcd827f5c3cf7ab</uketdterms:checksum>
   <uketdterms:embargodate>2026-06-30</uketdterms:embargodate>

   <dc:identifier xsi:type="dcterms:URI">http://localhost:4000/bitstreams/ae291418-a743-4898-8446-5d60085bfc34/download</dc:identifier>
   <uketdterms:checksum xsi:type="uketdterms:MD5">c72005c3f084da01050c00ff2ceeb207</uketdterms:checksum>
   <uketdterms:embargodate>2030-06-30</uketdterms:embargodate>
  
 <dcterms:license>http://localhost:4000/bitstreams/06050a11-6f54-49c0-ab40-d3a0411f315b/download</dcterms:license>
   <uketdterms:checksum xsi:type="uketdterms:MD5">eb9d2d75cbfff5683a827dd2229be5c0</uketdterms:checksum>
   <dc:rights>https://creativecommons.org/licenses/by/4.0/</dc:rights>
   <dc:subject>test</dc:subject>`

**Comment by bram-atmire** [comment]
This might be a tangent, and I'm not sure how much it's going to help you with UKETD/EthOS compliance, but I quite like the approach that RIOXXv3 is taking on this with the relation elements & the embargo dates:

https://www.rioxx.net/profiles/v3-0-rc-1/#dc:relation

```
<dc:relation type="https://schema.org/ScholarlyArticle" deposit_date="2021-07-28" resource_exposed_date="2021-07-28">
    https://www.repository.org/article_1234567.pdf
</dc:relation>
```

**Comment by amgciadev** [comment]
> This might be a tangent, and I'm not sure how much it's going to help you with UKETD/EthOS compliance, but I quite like the approach that RIOXXv3 is taking on this with the relation elements & the embargo dates:
> 
> https://www.rioxx.net/profiles/v3-0-rc-1/#dc:relation
> 
> ```
> <dc:relation type="https://schema.org/ScholarlyArticle" deposit_date="2021-07-28" resource_exposed_date="2021-07-28">
>     https://www.repository.org/article_1234567.pdf
> </dc:relation>
> ```

Thank you @bram-atmire for this! This looks like it will be very useful for the RIOXX OAI endpoint. Once version 3 of the schema is released, we could simply add support for it in the rioxx crosswalk, provided that the embargo information is available in the oai metadata via the implementation in this PR, if accepted.

**Review by paulo-graca — COMMENTED** [review]
Thank you @amgciadev for this contribution. I think it will fixed the reported issue (I didn't tried yet). I would like to add that 4Science did an important contribution to XOAI that enabled us to extend OAI by creating plugins. Meaning that, if you are interested in adding nonexisting XML to XOAI you can just add a new plugin, a configurable one that can be enabled or disabled in xoai Spring configuration.

Please check this plugin I've added:
https://github.com/DSpace/DSpace/blob/main/dspace-oai/src/main/java/org/dspace/xoai/app/plugins/AccessStatusElementItemCompilePlugin.java

I would also add that Access Status already adds item's access status. You can determine if something is an embargo, It performs a calculation based on this:
https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java#L107-L159

the logic of it can we reused as much as possible. They already can determine and differentiate if it's an embargo, or restricted. This means, if you use it, all will be coherent. 
Looking at it closer, the only thing it's missing are the resource policy dates, if Access Status could be refactored to return those dates, we could have a much better solution that could also consider embargo dates that are specified in resource policies.

**Comment by amgciadev** [comment]
Hi @paulo-graca, thanks so much for looking at reviewing this, and for the suggestions - I quite like the idea of reusing much of what already there for AccessStatus. I did look at it at first but didn't think to extend it. Thanks so much for the pointer!

**Comment by amgciadev** [comment]
Hi @paulo-graca and @tdonohue, just to give you some updates and thoughts here:

- I've looked into @paulo-graca suggestion of refactoring the AccessStatus classes to include a method that returns embargo information following the **same principles and logic** of the current **DefaultAccessStatusHelper** in that it only looks at either the primary bitstream or the first bitstream otherwise. This is easy to implement and expose in OAI via AccessStatusElementItemCompilePlugin.
- However, to fulfil this specific use case for the UKETD profile, the key requirement is to expose embargo dates for all bitstreams in the ORIGINAL (default content bundles), hence it felt simpler and more appropriate to do this via **org.dspace.xoai.util.ItemUtils** as that is the current place where the information about bitstreams for a given item is generated.

This is why I feel the current, simpler implementation is more appropriate, and can be re-used in any other profile that populates information about content bitstreams. Looking forward to your thoughts.

**Comment by amgciadev** [comment]
@paulo-graca, after some more thinking, I have decided to expand the DefaultAccessStatusHelper functionality to expose an embargo date following the same logic as that for calculating the access status: if the primary bitstream is present locate an active embargo for it, otherwise use the first bitstream available. I have also updated the relevant unit tests to include relevant embargo checks.

I have kept the addition to ItemUtils to continue to expose embargo dates for bitstreams in the original bundle via the bitstreams metadata to also support a future use case where these also need to be exposed in OAI, independently of the access status and for all bitstreams. One use case for this would be supporting the RIOXX metadata profile as @bram-atmire  has suggested earlier in this trail.

**Review by paulo-graca — COMMENTED** [review]
Thank you @amgciadev !
I just reviewed your PR. I have an implementation concern related with the current approach. Nothing that we can't refactor later on. But I would like to have feedback from others, like the original commiters or reviewers (@tdonohue ). 

My major concern with this approach is that we are bringing embargo retrieving methods to a level that doesn't make sense to me. The way to setup embargo is configurable using Spring configurations - https://github.com/DSpace/DSpace/blob/main/dspace/config/spring/api/access-conditions.xml

```xml
    <bean id="embargoed" class="org.dspace.submit.model.AccessConditionOption">
        <property name="groupName" value="Anonymous"/>
        <property name="name" value="embargo"/>
        <property name="hasStartDate" value="true"/>
        <property name="startDateLimit" value="+36MONTHS"/>
        <property name="hasEndDate" value="false"/>
    </bean>
```
This configuration can/could change. 
My suggestion is to change the way the embargo is obtained in this implementation to retrieve generic data like dates and changing your plugin to know how to handle with that data, since it's limited to embargo, could deal with that at that level.

As I said I would like to have feedback from others on this.

File: dspace-api/src/main/java/org/dspace/access/status/AccessStatusHelper.java
Comment (tdonohue) [reply dspace-api/src/main/java/org/dspace/access/status/AccessStatusHelper.java:39]: @paulo-graca : I'm personally OK with this approach because of how `DefaultAccessStatusHelper` is currently implemented.

While I *agree* it'd be nice if this AccessStatusHelper (and DefaultAccessStatusHelper) was aligned with the configurations in `access-conditions.xml`, currently it's entirely separate.  It ignores the configurations and applies it's own "access status" labels based entirely on our "State Definitions" at https://wiki.lyrasis.org/display/DSDOC7x/DSpace+Item+State+Definitions

So, this PR follows that same approach.. it's not very "generic", but it seems defined based on how we define an "embargo" works in DSpace.

File: dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java:213]: To better align with how embargoes work, I think we need to change this to `retrieveShortestEmbargo`.

If somehow you have two Anonymous policies with different embargo dates (which should be very rare), once the **earliest embargo date passes**, that Item will be publicly accessible.  

So, in this scenario, if we return the *longest Embargo*, I believe that would result in an incorrect embargo date.

File: dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java
Comment (tdonohue) [reply dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java:228]: I think this assumption is necessary because it's how the general idea of an embargo is defined in DSpace: https://wiki.lyrasis.org/display/DSDOC7x/DSpace+Item+State+Definitions

So, the behavior of this PR is correct.  An embargo always has a start date.   

As @paulo-graca points out though, it's possible to create an "access condition" named `embargo` that *doesn't have a start date*.  In that situation though the name `embargo` is just a name... it doesn't change the behavior of embargoes in DSpace's UI/REST API.  

In the UI/REST API, an Item/Bitstream will appear embargoed whenever it has a policy which has a `start date` in the future.  That's hardcoded into how DSpace works, and isn't actually configurable.  

(The `access-condition.xml` is only used right now in the Submission / Batch Editing UIs to provide an easier way to configure access settings...but those configs don't impact how DSpace behaves with respect to embargoes or leases.)

File: dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java
Comment (tdonohue) [reply dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java:178]: I *think* @paulo-graca is correct.  The embargo date logic has moved to `DefaultAccessStatusHelper`.. So I think all your customizations to `ItemUtils` should be removed?  (Unless I'm misunderstand how this all works)

**Review by tdonohue — CHANGES_REQUESTED** [review]
@amgciadev ; Overall, this is looking good to me.  

First, I've verified it works by using the `uketd_dc` metadata format and creating an Item with `dc.type=Thesis` (required by uketd_dc) with an embargoed bitstream.  After doing so, I can see the Item listed at http://localhost:8080/server/oai/request?verb=ListRecords&metadataPrefix=uketd_dc with a correct `<uketdterms:embargodate>`.

That said, I do have a few comments inline (some also in response to @paulo-graca's questions) & recommend some code cleanup.  I think the basic logic is fine, but there's some unnecessary / slightly incorrect code in this PR still.

**Comment by amgciadev** [comment]
Hi @paulo-graca and @tdonohue, first of all many thanks for taking the time to review this PR and the very useful feedback.

I will incorporate the suggested changes related to the slightly incorrect current logic that looks at longest embargo. I wanted to clarify in relation to the customisation to ItemUtils: this customisation is different as it allows to make embargo data present on all files associated with a given item so that is available if needed by any configured OAI crosswalk, whereas the current approach of AccessStatus only retrieves one embargo, based on the configured logic. This does not account for cases where one needs to know embargoes on all files within an item and there are cases wheres different files have different embargoes intentionally.

File: dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java
Comment (amgciadev) [reply dspace-api/src/main/java/org/dspace/access/status/DefaultAccessStatusHelper.java]: Done now

File: dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java
Comment (amgciadev) [reply dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java:178]: I think it still is as it serves a different case where embargo information is available for all bitstreams, whereas access status only provides one embargo, based on the configured logic

**Review by paulo-graca — COMMENTED** [review]
Thank you @amgciadev for addressing my feedback. Sorry if I'm taking your time with this. I just think we could have a solution that could not only address embargo situations, but with a little extension, it could also address later functionalities in the future without compromising what you are currently doing and proposing.

**Comment by paulo-graca** [comment]
> or, if you want to extract embargoed dates using XPATH, you could use proper XSL templates for that. You could just expose what's currently on the resource policy.

My intend it's to have just the resource policy data, just
``` xml
<element name="bitstreams">
   <element name="bitstream">
     <field name="name">2019_ENSPQ_AOliveira_Poster_R.pdf</field>
     <element name="resourcePolicies"> 
        <element name="resourcePolicy">
              <field name="group">ANONYMOUS</field>
              <field name="user"></field>
              <field name="action">READ</field>
              <field name="start-date">2023/09/14</field>
              <field name="end-date">2023/09/20</field>
        </element>
     </element>
   </element>
</element>
```

and let XSLT and XPATH define what is the embargo (group=ANONYMOUS + action=READ + START_DATE)

**Comment by amgciadev** [comment]
> > or, if you want to extract embargoed dates using XPATH, you could use proper XSL templates for that. You could just expose what's currently on the resource policy.
> 
> My intend it's to have just the resource policy data, just
> 
> ```
> <element name="bitstreams">
>    <element name="bitstream">
>      <field name="name">2019_ENSPQ_AOliveira_Poster_R.pdf</field>
>      <element name="resourcePolicies"> 
>         <element name="resourcePolicy">
>               <field name="group">ANONYMOUS</field>
>               <field name="user"></field>
>               <field name="action">READ</field>
>               <field name="start-date">2023/09/14</field>
>               <field name="end-date">2023/09/20</field>
>         </element>
>      </element>
>    </element>
> </element>
> ```
> 
> and let XSLT and XPATH define what is the embargo (group=ANONYMOUS + action=READ + START_DATE)

I like this approach and more than happy to implement it in this way. I wanted to also thank you for the really useful suggestions, and it is not a problem at all to make changes that will make the solution more generic and reusable/extensible in the future.

**Comment by amgciadev** [comment]
@paulo-graca re:

> Your method will be executed for every bitstream of every bundle of every public item. We will add a considerable overhead to OAI processing. Doing so, I think your method should be changed to enable (in the future) us to expose more data to XOAI.

Thanks for spotting this and the potential performance issue. I wanted to suggest that perhaps is useful then to only look at the ORIGINAL bundle given that files in that bundle are the only ones where access controls apply (and are intended for public display) @tdonohue what are your thoughts?

**Comment by tdonohue** [comment]
@amgciadev & @paulo-graca : While I'm not against the suggestions above, it's unclear to me when/how this resource policy/embargo data would be used *bitstream by bitstream*.  Currently, this PR only modifies the `uketd_dc` format, and that only uses the embargo at the *item level* (available in the "access-status").

So, I'm a little worried we are building something into `ItemUtils` that doesn't have a clear use case.  When would we need to return the resource-policy / embargo information for *every bitstream* via OAI-PMH?   If there's a clear use case for that, I'm OK with it (and in that scenario, I prefer the approach @paulo-graca [notes above](https://github.com/DSpace/DSpace/pull/8926#issuecomment-1719530960)).  

However, I'm unaware of any OAI-PMH crosswalk/metadata format that requires that level of detail.  So, I'm worried we are adding unnecessary detail into XOAI (adding overhead to its processing) which currently is unused.

**Comment by tdonohue** [comment]
NOTE: The Integration Test (IT) failures in this PR are coming from this separate bug: #9068   I'm working on getting a reviewer on that other PR to fix these failures.  Currently these IT failures are seemingly random, but they are happening in this PR very consistently.

This is a long way of saying, disregard the IT failures in this PR.  They are *not* caused by the code in this PR.  Once #9068 is merged, we should ensure this PR gets updated to the latest code, as which point the IT failures should disappear.

**Comment by tdonohue** [comment]
UPDATE: The random IT failure issue described in #9068 should now be fixed on `main` (as that PR was merged).  This PR should be updated to latest `main` in order to ensure the ITs all succeed.

**Comment by paulo-graca** [comment]
> @amgciadev & @paulo-graca : While I'm not against the suggestions above, it's unclear to me when/how this resource policy/embargo data would be used _bitstream by bitstream_. Currently, this PR only modifies the `uketd_dc` format, and that only uses the embargo at the _item level_ (available in the "access-status").
> 
> So, I'm a little worried we are building something into `ItemUtils` that doesn't have a clear use case. When would we need to return the resource-policy / embargo information for _every bitstream_ via OAI-PMH? If there's a clear use case for that, I'm OK with it (and in that scenario, I prefer the approach @paulo-graca [notes above](https://github.com/DSpace/DSpace/pull/8926#issuecomment-1719530960)).
> 
> However, I'm unaware of any OAI-PMH crosswalk/metadata format that requires that level of detail. So, I'm worried we are adding unnecessary detail into XOAI (adding overhead to its processing) which currently is unused.

Thank you @tdonohue! I think we have a real use case. OpenAIRE's schema - https://openaire-guidelines-for-literature-repository-managers.readthedocs.io/en/v4.0.0/field_filelocation.html  - foresees that every file could have a specific access right. In our current implementation we are assuming that every file in the item "drinks" his access rights from the defined access right calculated for the Item (work done with Access Status). But that approach isn't totally accurate for every file, mainly because we could have a different resource policy for the second or third file.
Another use case could be what @bram-atmire pointed out. Other schemas could also benefit from this level in detail from XOAI. 

Perhaps, and before we discard the approach, and the extra overhead, we could hear others like @benbosman or @abollini that work in local and international DSpace projects. What is their sensibility.
Just to point out that the overhead will be felt only at the XOAI indexing level. Something that as to be executed only from the CLI:
https://wiki.lyrasis.org/display/DSDOC7x/OAI#OAI-OAI-PMHServerMaintenance
we will experience more time for XOAI processing but that shouldn't affect UI experience.

**Comment by paulo-graca** [comment]
I also think we could have that discussion in a separate PR, I'm not interested in blocking this work until a decision is made. I will create the issue.

**Comment by amgciadev** [comment]
> I also think we could have that discussion in a separate PR, I'm not interested in blocking this work until a decision is made. I will create the issue.

Can you add me to that?

**Review by paulo-graca — COMMENTED** [review]
I was able to perform a test using:
```
/dspace/bin/dspace oai import -c
```

without errors 
```
OAI 2.0 manager action started
Clearing index
Index cleared
Using full import.
Full import
1000 items imported so far...
Total: 1037 items
Total: 1 items
Purging cached OAI responses.
OAI 2.0 manager action ended. It took 108 seconds.
```

and while checking Solr I was able to see the new expected embargo's entry:
``` xml
  <element name="others">
    <field name="handle">10000.99/888</field>
    <field name="identifier">oai:example.org:10000.99/888</field>
    <field name="lastModifyDate">2018-09-05 03:00:33.32</field>
    <element name="access-status">
      <field name="value">restricted</field>
      <field name="embargo">9999-01-01</field>
    </element>
  </element>
```

but with a small contradiction, the item is with `restricted access` type (because the mechanism assumes that if a certain start date is greater than a configured value, for instance, the year 9999 https://github.com/DSpace/DSpace/blob/main/dspace/config/dspace.cfg#L844-L853 - it should be of the restricted type), but the shown date is for the `embargo` type, when we don't have embargos.

Shouldn't we consider a start-date instead?
```
      <field name="start-date">9999-01-01</field> 
```

**Comment by amgciadev** [comment]
@paulo-graca I am not sure it is fully a contradiction, because the item status is "restricted" but until 9999-01-01 which here is considered to be a "permanent embargo" but it is still an embargo.

**Comment by paulo-graca** [comment]
Thank you @amgciadev for your reply. But, if the system doesn't consider it as an embargo, why should we keep the embargo date? 

My only afraid is that someone could just simply consider the 

``` xpath
element[@name="others"]/element[@name="access-status"]/field[@name=embargo]
```
in the transformations to obtain the date without taking into account the value of `access-status` itself. Leaving to expose misleading metadata.

@tdonohue if this isn't such a big issue for you, this PR has a +1 from my end. It can be merged.

**Comment by amgciadev** [comment]
Thanks @paulo-graca. I think the system does consider it as an embargo: it is considered a permanent one from the system's perspective, and the access status label is "restricted" instead. This also aligns with the current access options through the UI's "Access controls" section of the Edit iteem, where one would use the "Embargo" access level to specify a permanent embargo. @tdonohue  your thoughts can help clarify here! :)

**Comment by paulo-graca** [comment]
**restricted** and [embargoed](https://vocabularies.coar-repositories.org/access_rights/) by definition are two different things. 

Could you please check:
https://vocabularies.coar-repositories.org/access_rights/

> **embargoed access**
> Embargoed access refers to a resource that is metadata only access until released for open access on a certain date. Embargoes can be required by publishers and funders policies, or set by the author (e.g such as in the case of theses and dissertations).

> **restricted access**
> Restricted access refers to a resource that is available in a system but with some type of restriction for full open access. This type of access can occur in a number of different situations. Some examples are described below: The user must log-in to the system in order to access the resource The user must send an email to the author or system administrator to access the resource Access to the resource is restricted to a specific community (e.g. limited to a university community)


This topic isn't totally my beach, but my assumption is that when the resource is at Embargoed Access state, it's hoped to eventually have access to it in a lifetime (after embargo it will be in open access). With the Open Access perspective it will be tried to define limits to what and for how long the resource can be embargoed. For instance, here, in my country, funded doctoral thesis can have a maximum of 12 months of embargoed period (defined by a national policy). But aren't allowed to be restricted. If one item is set with embargo until the year date 9999, this means that the submitter doesn't want that item to be publicly available, meaning that item should be restricted. DSpace assumes this through forever embargo (based on a configurable date) and exposes the item as being restricted. So, for those cases, I'm not sure if we should expose an embargo date for when the item shall be opened, because it will "never" be (not in our lifetime).

**Comment by amgciadev** [comment]
@paulo-graca thanks for the COAR definitions and I agree with the differences between embargo and restricted. The issue is that the current system implementation does not fully account for the difference between these two scenarios:

- Restricted where there is no start date set, i.e. the resource has an administrative policy which corresponds with DSpace "Administrator" access control level
- Restricted where the access control is set in the system as "Embargo" with a date that meets the configured forever embargo criteria

At the moment the system does not differentiate between those two possibilities which are both classed as "Restricted".

I am not disagreeing with what you are saying, I am simply saying that the date exposed is still an embargo date, which happens to be indefinite.

**Comment by tdonohue** [comment]
@amgciadev and @paulo-graca :  I think this could be solved with just a few lines of code. If I understand correctly, we could just add a check for "embargo" status to `getEmbargoFromItem(Context, Item)`.  Something like this:

```
 @Override
 public String getEmbargoFromItem(Context context, Item item)
            throws SQLException {

            // If Item status is not "embargo" then return a null embargo date.
            String accessStatus = getAccessStatusFromItem(context, item)
            if (!accessStatus.equals(EMBARGO)) {
               return null;
            }
```

Overall, I feel this seems like a rare occurrence.  But, I see the point that we may want to first ensure the Access Status is confirmed to be an "embargo" before returning an embargo date.  It is a bit odd if we have an embargo date value returned for non-embargo status types.

**Review by paulo-graca — APPROVED** [review]
Thank you again @amgciadev for this contribution!

**Comment by doana** [comment]
> Thank you @tdonohue! I think we have a real use case. OpenAIRE's schema - https://openaire-guidelines-for-literature-repository-managers.readthedocs.io/en/v4.0.0/field_filelocation.html - foresees that every file could have a specific access right. In our current implementation we are assuming that every file in the item "drinks" his access rights from the defined access right calculated for the Item (work done with Access Status). But that approach isn't totally accurate for every file, mainly because we could have a different resource policy for the second or third file. Another use case could be what @bram-atmire pointed out. Other schemas could also benefit from this level in detail from XOAI.

Yes this OpenAIRE schema context is exactly where I've run across this issue. We're working on OpenAIRE compliance at the moment but ran up against this. Currently our OAI response isn't populating the oaire: file accessRightsURI attribute properly. As things stand now, the path forward involves adding dc.rights metadata to each item so that it populates the accessRightsURI attribute. But that strikes me as a problematic approach because we might end up incorrectly reporting a bitstream as open access, for example, when the bitstream actually is under some other access policy. It also means that we have to take on the work of managing those values across the whole repository. That's the kind of work that could easily get missed and put our records in an inconsistent state. 

IMO a little extra overhead in the XOAI indexing would be worth the finer level of granularity it would enable, given that we can run this indexing process during off-peak hours.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @amgciadev .  This looks good to me now too!

**Pull request merged**
